File: docs/legacy(old PACE)/C#/PaceAPISample/sample/JobPlanRefreshSample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;

namespace Pace_Web_Service_SDK.sample
{
    class JobPlanRefreshSample
    {
        private static CreateObjectPortType createObjectHttpBinding = PaceClient.getCreateObjectHttpBinding();
        private static ReadObjectPortType readObjectHttpBinding = PaceClient.getReadObjectHttpBinding();
        private static InvokeProcessPortType invokeProcessHttpBinding = PaceClient.getInvokeProcessHttpBinding();

        public static void Run()
        {
            Job job;
            Console.WriteLine("Enter Y if you want to create a sample job from the item template ");
            if( "Y".Equals(Console.ReadLine()))
            {
                String itemTemp = SDK.readStringInputFromTheUser( "Please enter item template name to be created : " );
                ItemTemplate template = ItemTemplateSample.createItemTemplate( itemTemp );
                job = ItemTemplateSample.createJob(template);
            }
            else
            {
                job = new Job();
                Console.WriteLine("Enter the job number");
                job.job = Console.ReadLine();
                job = readObjectHttpBinding.readJob(job);
            }

            createSampleJobCosts( job.job, "01" );

            findJobPlans( job.job, "01" );
            invokeProcessHttpBinding.refreshJobPlansForJob( job );
            Console.WriteLine("Job Plans refreshed");
            invokeProcessHttpBinding.updateLinksJobPlansForJob(job);
            Console.WriteLine("Links Updated");
            findJobPlans( job.job, "01" );
        }

        public static void createSampleJobCosts(String jobId, String partId)
        {
            JobCost jobCost = new JobCost();
            jobCost.job = jobId;
            jobCost.jobPart = partId;

            jobCost.activityCode = "000";
            jobCost.chargeClass = 1;

            jobCost = createObjectHttpBinding.createJobCost(jobCost);
        }

        public static void findJobPlans(String jobId, String partId)
        {
            FindObjectsPortType findObjectsHttpBinding = PaceClient.getFindObjectsHttpBinding();
            ArrayOfString jobPlanIds = findObjectsHttpBinding.find("JobPlan", "@job='" + jobId + "'");
            Console.WriteLine(jobPlanIds.Count + " job plans for job " + jobId);
        }
    }
}

File: docs/legacy(old PACE)/java/src/main/com/pace2020/epace/sdk/sample/JobPlanRefreshSample.java
package com.pace2020.epace.sdk.sample;

import java.util.ArrayList;
import java.util.List;

import com.pace2020.epace.object.ItemTemplate;
import com.pace2020.epace.object.Job;
import com.pace2020.epace.object.JobCost;
import com.pace2020.epace.object.JobPart;
import com.pace2020.epace.sdk.util.SecuredWSDLClient;

/**
 * @author <a href="mailto:faizanr@efi.com">faizan raza</a>
 */
public class JobPlanRefreshSample extends SecuredWSDLClient
{
    public static void main( final String[] args ) throws Exception
    {
        new JobPlanRefreshSample().run( args );
    }

    public void run( final String[] args ) throws Exception
    {
        Job job;
        if( args.length == 0 )
        {
            final CreateTwoJobsUsingItemTemplatesAndCreateComboJobSample
                sample = new CreateTwoJobsUsingItemTemplatesAndCreateComboJobSample("HOUSE", "12345");
            final ItemTemplate template = sample.getOrCreateItemTemplate( "TMP" );

            final List<JobPart> parts = new ArrayList<>();

            job = sample.createJob( template, parts );
        }
        else
        {
            job = new Job();
            job.setJob( args[0] );
            job = getReadObjectPortType().readJob( job );
        }

        createSampleJobCosts( job.getJob(), "01" );

        findJobPlans( job.getJob(), "01" );
        getInvokeProcessHttpPort().refreshJobPlansForJob( job );
        System.out.println( "Job plans refreshed...." );
        getInvokeProcessHttpPort().updateLinksJobPlansForJob( job );
        System.out.println( "Links Updated...." );
        findJobPlans( job.getJob(), "01" );
    }

    public void createSampleJobCosts( final String jobId, final String partId )
    {
        JobCost jobCost = new JobCost();
        jobCost.setJob( jobId );
        jobCost.setJobPart( partId );

        jobCost.setActivityCode( "000" );
        jobCost.setChargeClass( 1 );

        jobCost = getCreateObjectPortType().createJobCost( jobCost );
    }

    public void findJobPlans( final String jobId, final String partId )
    {
        final List<String> jobPlanIds =
            getFindObjectsPortType().find( "JobPlan", "@job='" + jobId + "'" ).getString();

        System.out.println( jobPlanIds.size() + " job plans for job " + jobId );
    }
}